crates/cxx-qt-gen/src/naming/rust.rs

```diff
@@ -3,112 +3,103 @@
 //
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
-use syn::{GenericArgument, PathArguments, PathSegment, ReturnType, Type, TypeReference};
+use syn::{
+    GenericArgument, PathArguments, PathSegment, Result, ReturnType, Type, TypePath, TypeReference,
+};
 
 use crate::naming::TypeNames;
 
+macro_rules! convert_elem {
+    ($id:ident, $variant:path, $type_names:ident) => {{
+        let mut ty = $id.clone();
+        *ty.elem = syn_type_cxx_bridge_to_qualified(&ty.elem, $type_names)?;
+        Ok($variant(ty))
+    }};
+}
+
+fn qualify_type_path(ty_path: &TypePath, type_names: &TypeNames) -> Result<Type> {
+    let mut ty_path = ty_path.clone();
+
+    // Convert any generic arguments
+    for segment in ty_path.path.segments.iter_mut() {
+        if let PathArguments::AngleBracketed(angled) = &mut segment.arguments {
+            for arg in angled.args.iter_mut() {
+                if let GenericArgument::Type(ty) = arg {
+                    *ty = syn_type_cxx_bridge_to_qualified(ty, type_names)?;
+                }
+            }
+        }
+    }
+
+    // Convert the first element if it matches
+    if let Some(segment) = ty_path.path.segments.first() {
+        let qualified_prefix = match segment.ident.to_string().as_str() {
+            // Note we need to fully qualify any types that CXX supports that aren't
+            // - primitive types https://doc.rust-lang.org/stable/std/primitive/index.html
+            // - prelude types https://doc.rust-lang.org/stable/std/prelude/index.html
+            //
+            // We could also fully qualify types primitive and prelude types for full macro hygiene.
+            "CxxString" | "CxxVector" | "SharedPtr" | "UniquePtr" | "WeakPtr" => Some(vec!["cxx"]),
```

Comment:
this is stuff that would be good in the separate lib

Reply:
but is that what naming has / is becoming what did we decide before ?

I thought we were thinking of having naming -> generic CXX helper lib

Reply:
If I recall correctly the idea is to eventually move the naming module into its own library.
So that's alright then?

Reply:
If we think that the naming module would not be dependant on CXX-Qt crate types, then i guess yup :thinking: